image.go

```diff
@@ -66,8 +67,18 @@ func (i *image) Unpack(ctx context.Context, snapshotterName string) error {
 	for _, layer := range layers {
 		unpacked, err := rootfs.ApplyLayer(ctx, layer, chain, sn, a)
 		if err != nil {
-			// TODO: possibly wait and retry if extraction of same chain id was in progress
-			return err
+			// wait and retry if extraction of same chain id was in progress
+			for retry := 3; retry > 0; retry-- {
+				// 3 second wait seems to be working for various diffrent image pull (parallel) tests.
+				time.Sleep(time.Second * 3)
+				unpacked, err = rootfs.ApplyLayer(ctx, layer, chain, sn, a)
+				if err == nil {
```

Comment:
Which error is being returned here? We should expect either an unavailable or already exists error.

Reply:
Yes its `already exist`

```
failed to prepare extraction layer: snapshot extract sha256:8aa4fcad5eeb286fe9696898d988dc85503c6392d1a2bd9023911fb0d6d27081 already exists: already exists
```